src/libs/address.js

```diff
@@ -75,3 +80,14 @@ function findAdminIdunn(raw, name) {
     .values(raw.address?.admins || {})
     .find(a => a.class_name === name);
 }
+
+/**
+ * Format administrative hierarchy where a poi (from Idunn or Bragi) is located
+ * @param {*} raw - the raw poi object
+ */
+function extractParentAdmins(raw) {
```

Comment:
Instead of returning a preformatted string, can we extract values independently ?
The city and country should already be in an address object, so we should only extract new information, like region/state.

This could be done in the `normalize` function, with helpers like `findAdminBragi`.

Reply:
This way, the formatting would still occurs at one place : in the `format` function

Reply:
I was a bit reluctant at first to deal with the complexity of all administrative regions in these functions.
But you're right, it's probably more future-proof to format the regions explicitly. 

I updated the PR to add 5 new fields in the address object, in order to cover all types of regions.